to_copy/usr/bin/pep-filter.sh

```diff
@@ -0,0 +1,268 @@
+        }
+        /sbin/ipset destroy "$sublist" \
+           || fail "Failed to delete ipset $sublist"
+    done
+
+    # delete a domain blacklist file
+    [ -f "$domain_file" ] && {
+       rm -f "$domain_file"
+       reload_filter
+    }
+    return 0
+}
+
+
+reset_filters() {
+    # delete all ipset lists and iptables rules
+    /sbin/ipset -n list | while read list_name; do
+        /sbin/iptables-save | grep -Eq -- "--match-set \<$list_name\>" && {     
+            /sbin/iptables -D ENCRYPTME -m set --match-set "$list_name" dst -j DROP \
+               || fail "Failed to delete iptables rule for the list $list_name"
+        }
+        /sbin/ipset destroy "$list_name" \
+           || fail "Failed to delete ipset $list_name"
+    done
+
+    # remove our domain blacklists
+    rm -rf "$FILTERS_DIR" \
+        || fail "Failed to delete domain lists"
+    reload_filter
+}
+
+
+# reads stdin to parse IPv4 CIDR ranges and domain names and filter them out
+append_list() {
+    local list_name="$1"
+    local cidr_file="$TMP_DIR/$list_name.cidr"
+    local domain_file="$TMP_DIR/$list_name.domains"
+    local stdin="$TMP_DIR/$list_name.stdin"
+
+    cat > "$stdin"
+    cat "$stdin" | grep -E "$CIDR_RE" > "$cidr_file"
+    cat "$stdin" | grep -E "$DOMAIN_RE" > "$domain_file"
+
+    [ -s "$cidr_file" ] &&  add_ips "$list_name" "$cidr_file"
+    [ -s "$domain_file" ] &&  add_domains "$list_name" "$domain_file"
+}
+
+
+unbound_health() {
+    unbound_process=`pgrep unbound`
+    # When `unbound` pid is not found, exit with 1, otherwise 0.
+    if [ -z "$unbound_process" ]; then
+        exit 1
+    fi
+    exit 0
+}
+
+filter_server_health() {
+    filter_server_process=`pgrep filter_server`
+    # When `filter_server` pid is not found, exit with 1, otherwise 0.
+    if [ -z "$filter_server_process" ]; then
+        exit 1
+    fi
+    exit 0
+}
+
+
+[ $# -ge 1 ] || {
+    usage
+    fail "No action given"
+}
+case "$1" in
+    append|replace|prune|reset|unbound_health|filter_server_health)
+        action="$1"
+        shift
+        ;;
+    *)
+        usage
+        fail "Invalid action: '$1'"
+esac
+
+
+[ "$action" = "append" ] && {
+    [ $# -eq 1 ] || fail "No list name given to append to"
+    list_name="$1" && shift
+    append_list "$list_name"
+}
+
+[ "$action" = "replace" ] && {
+    [ $# -eq 1 ] || fail "No list name given to replace"
+    list_name="$1" && shift
+    prune_list "$list_name"
+    append_list "$list_name"
+}
+
+[ "$action" = "prune" ] && {
+    [ $# -eq 1 ] || fail "No list name given to prune from"
+    list_name="$1" && shift
+    prune_list "$list_name"
+}
+
+[ "$action" = "reset" ] && {
+    reset_filters
+}
+
+[ "$action" = "unbound_health" ] && {
+    unbound_health
+}
+
+[ "$action" = "filter_server_health" ] && {
+    filter_server_health
+}
+
+# Ensure our IP tables are up-to-date so we can restore them on restart.
+/usr/sbin/iptables-save > /etc/encryptme/iptables.save \
```

Comment:
as per other comments, we don't want to use temp files for rules; instead we'll reconstruct based on the blacklist files during container startup

Reply:
This code was removed and a new function "reload" added to reconstruct using blacklist files on reboot